tests/test_magentic.py

```diff
@@ -21,23 +23,45 @@
 from tests.utils import _LogAssertion, format_magentic_function_args
 
 
-log10(openai)
+def _get_model_obj(llm_provider, model, params):
+    provider_map = {"openai": (OpenaiChatModel, log10(openai)), "anthropic": (AnthropicChatModel, log10(anthropic))}
```

Comment:
think `log10(openai)` and `log10(anthropic)` will be called during init of `provider_map`. if you are thinking to only call the one when it's needed, you may want to update the code a little bit. 
Otherwise, we could simply patch both in the beginning of the file.